app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,32 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription ID is provided,
+      # then only collect data for that subscription. Otherwise, collect all data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
```

Comment:
@djberg96 
Once #7895 is merged, any minute now, a small change needs to me made here to check if the EMS property `subscription` is set; if it is NIL then gather an array of all subscriptions otherwise use the `subscription` property value.
Should be as simple as:
- dropping the subscription_id parameter to the `gather_data_for_this_region` method
- rename `subscription_id` to `subscription`